src/ui/dfn-list.js

```diff
@@ -49,9 +49,34 @@ function show() {
           <a href="${`#${dfn.id}`}">
             ${dfn.textContent}
           </a>
+          ${labelDfnIfExported(dfn)} ${labelDfnIfUnused(dfn)}
         </li>
       `;
     });
   render`${definitionLinks}`;
   ui.freshModal(l10n.list_of_definitions, ul, button);
 }
+
+/**
+ * If a definition is exported, label it accordingly
+ * @param {HTMLElement} dfn a definition
+ */
+function labelDfnIfExported(dfn) {
+  if (dfn.hasAttribute("data-export")) {
+    return hyperHTML`<span class="exported">(exported)</span>`;
+  } else {
+    return null;
+  }
+}
+
+/**
+ * If a definition is unused, label it accordingly
+ * @param {HTMLElement} dfn a definition
+ */
+function labelDfnIfUnused(dfn) {
+  if (dfn.hasAttribute("definition-used")) {
```

Comment:
I would prefer if we do this test here only, and not in `core/link-to-dfn`, although it's a but more efficient to that way.
Why? We use `isUnused` here only, and a user may not even open the dfn-list, so it makes sense to keep that logic localized here (makes it hard to reason about) and not do unnecessary work in link-to-dfn.

We can do something like:
``` js
const isUnused = !document.querySelector(`a[href^="#${dfn.id}"]`);
if (isUnused) {
  return null;
}
return hyperHTML`<span class="unused">(unused)</span>`;
```

Reply:
@sidvishnoi do we expect all definitions in specs to have an `id`? 

* the sample spec in `basic.html` only has `<dfn>Foo</dfn>`.
* `<a>Foo</a>` results in `<a href="#dom-foo" class="internalDFN" data-link-type="idl"><code>Foo</code></a>` after respec has done its thing. 
* thus, using code above would mean that terms referenced in other parts of the spec would be incorrectly marked as `unused`

Reply:
> do we expect all definitions in specs to have an id?

Yes. If user doesn't specify an id, ReSpec auto-generates one.

> thus, using code above would mean that terms referenced in other parts of the spec would be incorrectly marked as unused

As UI is enabled after all plugins have processed, we are certain that ID generation and linking have been done already.

Reply:
@sidvishnoi ah i see the issue. XD your code should've been `if (isUnused) { return hyperHTML(unused) etc}` that's why my test started failing as well. XD